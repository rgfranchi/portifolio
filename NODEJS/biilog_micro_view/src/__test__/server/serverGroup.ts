import { Response } from "miragejs";
import {
  ISessionAccess,
  ISessionGroup,
} from "../../interfaces/SessionInterface";
const baseURL = "group";

export const groupRoutes = (actions: any) => {
  let headers = {};
  actions.get(`${baseURL}/findAll`, (schema: any) => {
    const dataBase = schema.db.groups;
    if (dataBase.length === 0) {
      return new Response(204, headers, "");
    }
    return new Response(200, headers, dataBase);
  });
  actions.get(`${baseURL}/listGroupSelect`, (schema: any) => {
    const dataBase: any = [];
    schema.db.groups.forEach((value: any) => {
      // @ts-ignore
      dataBase.push({ id: parseInt(value.id), nome: value.nome });
    });
    if (dataBase.length === 0) {
      return new Response(204, headers, "");
    }
    return new Response(200, headers, dataBase);
  });
};

const user: ISessionAccess = {
  label: "Usuário",
  page: {
    // leitura em ...Page.tsx
    create: {
      path: "/new",
      label: "Novo Usuário",
    },
    update: {
      path: "/update",
    },
    delete: {
      path: "/delete",
    },
  },
  path: "/user", // leitura em routes.tsx
};

const company: ISessionAccess = {
  path: "/company", // leitura em routes.tsx
  label: "Empresas",
  page: {
    // leitura em ...Page.tsx
    create: {
      path: "/new",
      label: "Nova Empresa",
    },
    update: {
      path: "/update",
    },
    delete: {
      path: "/delete",
    },
  },
};

const operator: ISessionAccess = {
  path: "/operator",
  label: "Operadores",
  page: {
    // leitura em ...Page.tsx
    create: {
      path: "/new",
      label: "Inserir Operador",
    },
    update: {
      path: "/update",
    },
    delete: {
      path: "/delete",
    },
  },
};

const forklift: ISessionAccess = {
  path: "/forklift",
  label: "Equipamentos",
  page: {
    // leitura em ...Page.tsx
    create: {
      path: "/new",
      label: "Inserir Equipamento",
    },
    update: {
      path: "/update",
    },
    delete: {
      path: "/delete",
    },
  },
};

const register: ISessionAccess = {
  label: "Cadastro",
  sub_item: [
    user,
    {
      path: "/machine",
      label: "Máquina",
      page: {
        create: {
          path: "/new",
          label: "Nova Máquina",
        },
        update: {
          path: "/update",
        },
        delete: {
          path: "/delete",
        },
      },
    },
  ],
};

/**
 * Valores MOCk para teste da entidade.
 */
export const listGroupMock: ISessionGroup[] = [
  {
    id: 1,
    nome: "Root",
    access: [company, operator, register, forklift],
  },
  {
    id: 2,
    nome: "Administrador",
    access: [company, operator],
  },
  {
    id: 3,
    nome: "Usuário Comum",
    access: [operator],
  },
];
