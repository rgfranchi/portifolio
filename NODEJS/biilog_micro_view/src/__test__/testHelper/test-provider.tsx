import React from "react";
import { MemoryRouter, Route, Router } from "react-router-dom";

import { objMockSession } from "./test-session";

import { Provider } from "react-redux";

//componentes de tela.
import { ModalLoading } from "../../components/ModalsComponent";

/**
 * Renderiza valores para teste.
 * Utilizando o render do framework @testing-library/react
 * Componente ModalLoading.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @returns
 */
export const renderProviderRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  // @ts-ignore
  function Wrapper({ children }) {
    return providerRouter(children, history, store, modal);
  }
  // @ts-ignore
  return render(ui, { wrapper: Wrapper });
};

/**
 * ConstrÃ³i com base no renderProviderRouter incluindo MemoryRouter e Router.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @param {*} initialEntries
 * @param {*} path
 * @returns
 */
export const renderProviderMemoryRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  initialEntries: any,
  path: string,
  modal: boolean = true
) => {
  const wrapper = (
    <MemoryRouter initialEntries={[initialEntries]}>
      <Route path={path}>{ui}</Route>
    </MemoryRouter>
  );
  return renderProviderRouter(render, wrapper, history, store, modal);
};

/**
 * Cria as funcionalidades do Provider e Router
 * Utilizando o mount do framework Enzyme
 * @param {*} mount
 * @param {*} children
 * @param {*} history
 * @param {*} store
 * @param {boolean} modal
 * @returns
 */
export const mountProviderRouter = async (
  mount: any,
  children: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  return await mount(providerRouter(children, history, store, modal));
};

export const createSessionCurrentUser = (store: any) => {
  store.dispatch({
    type: "session/currentUser",
    payload: objMockSession,
  });
};

const providerRouter = (
  children: any,
  history: any,
  store: any,
  modal: boolean
) => {
  return (
    <Provider store={store}>
      <Router history={history}>
        {modal && <ModalLoading />}
        {children}
      </Router>
    </Provider>
  );
};
