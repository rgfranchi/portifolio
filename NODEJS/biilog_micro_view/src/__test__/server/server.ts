import { createServer, RestSerializer, Model } from "miragejs";
import { userRoutes } from "./serverUser";
import { listCompanyMock, companyRoutes } from "./serverCompany";
import { listOperatorMock, operatorRoutes } from "./serverOperator";
import { listUserMock } from "./serverUser";
import { listGroupMock, groupRoutes } from "./serverGroup";
import { listForkliftMock, forkliftRoutes } from "./serverForklift";
import { listPlaceMock, placeRoutes } from "./serverPlace";

let isEmpty = false;
// variÃ¡vel para resposta vazia, impede alimentar com array original.
/**
 *
 * @param {boolean} empty
 */
export const forceEmpty = (empty: boolean) => {
  // console.info("ForceEmpty", empty);
  isEmpty = empty;
};

export const getIsEmpty = () => isEmpty;

// https://miragejs.com/docs/main-concepts/database/
// https://miragejs.com/api/classes/db-collection/
// https://miragejs.com/docs/main-concepts/route-handlers/
export const mockServer = () => {
  return createServer({
    serializers: {
      reminder: RestSerializer.extend({
        include: ["list"],
        embed: true,
      }),
    },
    models: {
      // userCurrent: Model,
      // user: Model,
      user: Model,
      company: Model,
      operator: Model,
      group: Model,
      forklift: Model,
      place: Model,
    },
    seeds(server) {
      // server.db.loadData({ userCurrent: isEmpty ? [] : userCurrentMock });
      server.db.loadData({ users: isEmpty ? [] : listUserMock });
      server.db.loadData({ groups: isEmpty ? [] : listGroupMock });
      server.db.loadData({ companies: isEmpty ? [] : listCompanyMock });
      server.db.loadData({ operators: isEmpty ? [] : listOperatorMock });
      server.db.loadData({ forklifts: isEmpty ? [] : listForkliftMock });
      server.db.loadData({ places: isEmpty ? [] : listPlaceMock });
    },
    routes() {
      this.namespace = "api/v0";
      userRoutes(this);
      companyRoutes(this);
      operatorRoutes(this);
      groupRoutes(this);
      forkliftRoutes(this);
      placeRoutes(this);
      this.passthrough("https://viacep.com.br/ws/**");
    },
  });
};
