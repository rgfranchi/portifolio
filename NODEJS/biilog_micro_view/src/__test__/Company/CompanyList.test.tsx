import React from "react";

import { act } from "react-dom/test-utils";
import Enzime, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

import { createMemoryHistory } from "history";

import store from "../../redux/store";

import { mountProviderRouter } from "../testHelper/test-provider";

import CompanyList from "../../Company/CompanyList";
import { listCompanyMock } from "../server/serverCompany";

Enzime.configure({ adapter: new Adapter() });

let wrapper: any = null;
let history: any = null;
const listMock = listCompanyMock;

beforeEach(() => {
  history = createMemoryHistory();
});

describe("CompanyList snapshot", () => {
  test("empty", async () => {
    history.entries[0].key = "EMPTY_01";
    await setWrapperDefault([]);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("with data", async () => {
    history.entries[0].key = "DATA_01";
    await setWrapperDefault(listMock);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("CompanyList actions", () => {
  const count = listMock.length;
  const arrPos = 2;

  test("update", async () => {
    await setWrapperDefault(listMock);
    history.entries[0].key = "ACTION_UPDATE";
    const action = wrapper.find("ButtonIconUpdate");
    expect(action.length).toEqual(count);
    action.at(arrPos).simulate("click", { button: 0 });
    expect(history.location.pathname).toBe(
      "/urlFake/update/" + listMock[arrPos].id
    );
  });
  test("delete", async () => {
    const mockFunctionDelete = jest.fn((x) => x);
    await setWrapperDefault(listMock, mockFunctionDelete);
    history.entries[0].key = "ACTION_DEL";
    const action = wrapper.find("ButtonIconDelete");
    expect(action.length).toEqual(count);
    action.at(arrPos).simulate("click", { button: 0 });
    expect(mockFunctionDelete.mock.results[0].value).toBe(listMock[arrPos].id);
  });
});

describe("CompanyList access", () => {
  const count = listMock.length;

  test("update denied", async () => {
    expect(count).not.toEqual(0);
    await setWrapperDefault(listMock, () => {}, { delete: {} });
    history.entries[0].key = "ACTION_UPDATE";
    const action = wrapper.find("ButtonIconUpdate");
    expect(action.length).toEqual(0);
  });
  test("delete denied", async () => {
    expect(count).not.toEqual(0);
    await setWrapperDefault(listMock, () => {}, { update: {} });
    history.entries[0].key = "ACTION_DEL";
    const action = wrapper.find("ButtonIconDelete");
    expect(action.length).toEqual(0);
  });
});

const setWrapperDefault = async (
  list: any,
  handleDelete: any = () => {},
  access: any = { update: {}, delete: {} }
) => {
  await act(async () => {
    wrapper = await mountProviderRouter(
      mount,
      <CompanyList
        baseUrl="/urlFake"
        access={access}
        list={list}
        handleDeleteOpen={handleDelete}
      />,
      history,
      store
    );
  });
};
