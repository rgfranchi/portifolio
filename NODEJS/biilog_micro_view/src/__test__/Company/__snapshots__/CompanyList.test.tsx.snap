// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyList snapshot empty 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "EMPTY_01",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "EMPTY_01",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ModalLoading>
      <Modal
        animation={false}
        autoFocus={true}
        backdrop={true}
        centered={true}
        dialogAs={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "ModalDialog",
            "render": [Function],
          }
        }
        enforceFocus={true}
        keyboard={true}
        onHide={[Function]}
        restoreFocus={true}
        show={false}
        size="lg"
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          containerClassName="modal-open"
          enforceFocus={true}
          keyboard={true}
          manager={
            BootstrapModalManager {
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "modals": Array [],
              "scrollbarSize": 0,
            }
          }
          onEnter={[Function]}
          onEntering={[Function]}
          onEscapeKeyDown={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onHide={[Function]}
          renderBackdrop={[Function]}
          renderDialog={[Function]}
          restoreFocus={true}
          show={false}
        />
      </Modal>
    </ModalLoading>
    <CompanyList
      access={
        Object {
          "delete": Object {},
          "update": Object {},
        }
      }
      baseUrl="/urlFake"
      handleDeleteOpen={[Function]}
      list={Array []}
    >
      <ForwardRef>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th>
                Nome
              </th>
              <th>
                Contato
              </th>
              <th>
                CNPJ
              </th>
              <th>
                E-mail
              </th>
              <th>
                Ações
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </ForwardRef>
    </CompanyList>
  </Router>
</Provider>
`;

exports[`CompanyList snapshot with data 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "DATA_01",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "DATA_01",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ModalLoading>
      <Modal
        animation={false}
        autoFocus={true}
        backdrop={true}
        centered={true}
        dialogAs={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "ModalDialog",
            "render": [Function],
          }
        }
        enforceFocus={true}
        keyboard={true}
        onHide={[Function]}
        restoreFocus={true}
        show={false}
        size="lg"
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          containerClassName="modal-open"
          enforceFocus={true}
          keyboard={true}
          manager={
            BootstrapModalManager {
              "containers": Array [],
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "modals": Array [],
              "scrollbarSize": 0,
            }
          }
          onEnter={[Function]}
          onEntering={[Function]}
          onEscapeKeyDown={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onHide={[Function]}
          renderBackdrop={[Function]}
          renderDialog={[Function]}
          restoreFocus={true}
          show={false}
        />
      </Modal>
    </ModalLoading>
    <CompanyList
      access={
        Object {
          "delete": Object {},
          "update": Object {},
        }
      }
      baseUrl="/urlFake"
      handleDeleteOpen={[Function]}
      list={
        Array [
          Object {
            "bairro": "Brooklin Paulista",
            "cidade": "São Paulo",
            "cnpj": "39.282.099/0001-05",
            "codPostal": "04557-030",
            "complemento": "Sala 2",
            "email": "edmondo.gazzillo@gmail.com",
            "estado": "São Paulo",
            "id": 1,
            "logradouro": "Rua Marquês de Cascais",
            "nome": "BIILOG",
            "nomeContato": "Edmondo Gazzillo",
            "numero": "152",
            "pais": "Brasil",
            "telefone1": "1132051654",
            "telefone2": "11996135016",
          },
          Object {
            "bairro": "Civit II",
            "cidade": "Serra",
            "cnpj": "03.640.467/0001-94",
            "codPostal": "29168-090",
            "complemento": null,
            "email": "fabioferreira@wine.com.br",
            "estado": "Espirito Santo",
            "id": 2,
            "logradouro": "Rua Comendador Alcídes Simão Helou",
            "nome": "WINE",
            "nomeContato": "Fábio Ferreira",
            "numero": "1478",
            "pais": "Brasil",
            "telefone1": "11968574521",
            "telefone2": null,
          },
          Object {
            "bairro": "Vila Aeroporto",
            "cidade": "Guarulhos",
            "cnpj": "78.803.809/0001-49",
            "codPostal": "07170-350",
            "complemento": null,
            "email": "vander_levorato@lincolnelectric.com.br",
            "estado": "São Paulo",
            "id": 3,
            "logradouro": "Av. Papa João Paulo I",
            "nome": "LinconLN",
            "nomeContato": "Vander Levorato",
            "numero": "1818",
            "pais": "Brasil",
            "telefone1": "1135284700 ",
            "telefone2": null,
          },
          Object {
            "bairro": "Éden",
            "cidade": "Sorocaba",
            "cnpj": "80.783.926/0001-30",
            "codPostal": "18087-220",
            "complemento": null,
            "email": "servicos.rental@brasif.com.br",
            "estado": "São Paulo",
            "id": 4,
            "logradouro": "Avenida Jerome Case",
            "nome": "CNHi",
            "nomeContato": "Não Informado",
            "numero": "1801",
            "pais": "Brasil",
            "telefone1": "1533341700",
            "telefone2": null,
          },
        ]
      }
    >
      <ForwardRef>
        <table
          className="table"
        >
          <thead>
            <tr>
              <th>
                Nome
              </th>
              <th>
                Contato
              </th>
              <th>
                CNPJ
              </th>
              <th>
                E-mail
              </th>
              <th>
                Ações
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              key="1"
            >
              <td>
                BIILOG
              </td>
              <td>
                Edmondo Gazzillo
              </td>
              <td>
                39.282.099/0001-05
              </td>
              <td>
                edmondo.gazzillo@gmail.com
              </td>
              <td>
                <ButtonIconUpdate
                  linkTo="/urlFake/update/1"
                >
                  <Button
                    active={false}
                    as={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "Link",
                        "propTypes": Object {
                          "innerRef": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    size="sm"
                    to="/urlFake/update/1"
                    variant="primary"
                  >
                    <Link
                      className="btn btn-primary btn-sm"
                      disabled={false}
                      to="/urlFake/update/1"
                    >
                      <LinkAnchor
                        className="btn btn-primary btn-sm"
                        disabled={false}
                        href="/urlFake/update/1"
                        navigate={[Function]}
                      >
                        <a
                          className="btn btn-primary btn-sm"
                          disabled={false}
                          href="/urlFake/update/1"
                          onClick={[Function]}
                        >
                          <FiEdit>
                            <IconBase
                              attr={
                                Object {
                                  "fill": "none",
                                  "stroke": "currentColor",
                                  "strokeLinecap": "round",
                                  "strokeLinejoin": "round",
                                  "strokeWidth": "2",
                                  "viewBox": "0 0 24 24",
                                }
                              }
                            >
                              <svg
                                fill="none"
                                height="1em"
                                stroke="currentColor"
                                strokeLinecap="round"
                                strokeLinejoin="round"
                                strokeWidth="2"
                                style={
                                  Object {
                                    "color": undefined,
                                  }
                                }
                                viewBox="0 0 24 24"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                                  key="0"
                                />
                                <path
                                  d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                                  key="1"
                                />
                              </svg>
                            </IconBase>
                          </FiEdit>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </Button>
                </ButtonIconUpdate>
                 
                <ButtonIconDelete
                  execute={[Function]}
                >
                  <Button
                    active={false}
                    disabled={false}
                    onClick={[Function]}
                    size="sm"
                    variant="danger"
                  >
                    <button
                      className="btn btn-danger btn-sm"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <AiFillDelete>
                        <IconBase
                          attr={
                            Object {
                              "viewBox": "0 0 1024 1024",
                            }
                          }
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 1024 1024"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                              key="0"
                            />
                          </svg>
                        </IconBase>
                      </AiFillDelete>
                    </button>
                  </Button>
                </ButtonIconDelete>
              </td>
            </tr>
            <tr
              key="2"
            >
              <td>
                WINE
              </td>
              <td>
                Fábio Ferreira
              </td>
              <td>
                03.640.467/0001-94
              </td>
              <td>
                fabioferreira@wine.com.br
              </td>
              <td>
                <ButtonIconUpdate
                  linkTo="/urlFake/update/2"
                >
                  <Button
                    active={false}
                    as={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "Link",
                        "propTypes": Object {
                          "innerRef": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    size="sm"
                    to="/urlFake/update/2"
                    variant="primary"
                  >
                    <Link
                      className="btn btn-primary btn-sm"
                      disabled={false}
                      to="/urlFake/update/2"
                    >
                      <LinkAnchor
                        className="btn btn-primary btn-sm"
                        disabled={false}
                        href="/urlFake/update/2"
                        navigate={[Function]}
                      >
                        <a
                          className="btn btn-primary btn-sm"
                          disabled={false}
                          href="/urlFake/update/2"
                          onClick={[Function]}
                        >
                          <FiEdit>
                            <IconBase
                              attr={
                                Object {
                                  "fill": "none",
                                  "stroke": "currentColor",
                                  "strokeLinecap": "round",
                                  "strokeLinejoin": "round",
                                  "strokeWidth": "2",
                                  "viewBox": "0 0 24 24",
                                }
                              }
                            >
                              <svg
                                fill="none"
                                height="1em"
                                stroke="currentColor"
                                strokeLinecap="round"
                                strokeLinejoin="round"
                                strokeWidth="2"
                                style={
                                  Object {
                                    "color": undefined,
                                  }
                                }
                                viewBox="0 0 24 24"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                                  key="0"
                                />
                                <path
                                  d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                                  key="1"
                                />
                              </svg>
                            </IconBase>
                          </FiEdit>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </Button>
                </ButtonIconUpdate>
                 
                <ButtonIconDelete
                  execute={[Function]}
                >
                  <Button
                    active={false}
                    disabled={false}
                    onClick={[Function]}
                    size="sm"
                    variant="danger"
                  >
                    <button
                      className="btn btn-danger btn-sm"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <AiFillDelete>
                        <IconBase
                          attr={
                            Object {
                              "viewBox": "0 0 1024 1024",
                            }
                          }
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 1024 1024"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                              key="0"
                            />
                          </svg>
                        </IconBase>
                      </AiFillDelete>
                    </button>
                  </Button>
                </ButtonIconDelete>
              </td>
            </tr>
            <tr
              key="3"
            >
              <td>
                LinconLN
              </td>
              <td>
                Vander Levorato
              </td>
              <td>
                78.803.809/0001-49
              </td>
              <td>
                vander_levorato@lincolnelectric.com.br
              </td>
              <td>
                <ButtonIconUpdate
                  linkTo="/urlFake/update/3"
                >
                  <Button
                    active={false}
                    as={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "Link",
                        "propTypes": Object {
                          "innerRef": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    size="sm"
                    to="/urlFake/update/3"
                    variant="primary"
                  >
                    <Link
                      className="btn btn-primary btn-sm"
                      disabled={false}
                      to="/urlFake/update/3"
                    >
                      <LinkAnchor
                        className="btn btn-primary btn-sm"
                        disabled={false}
                        href="/urlFake/update/3"
                        navigate={[Function]}
                      >
                        <a
                          className="btn btn-primary btn-sm"
                          disabled={false}
                          href="/urlFake/update/3"
                          onClick={[Function]}
                        >
                          <FiEdit>
                            <IconBase
                              attr={
                                Object {
                                  "fill": "none",
                                  "stroke": "currentColor",
                                  "strokeLinecap": "round",
                                  "strokeLinejoin": "round",
                                  "strokeWidth": "2",
                                  "viewBox": "0 0 24 24",
                                }
                              }
                            >
                              <svg
                                fill="none"
                                height="1em"
                                stroke="currentColor"
                                strokeLinecap="round"
                                strokeLinejoin="round"
                                strokeWidth="2"
                                style={
                                  Object {
                                    "color": undefined,
                                  }
                                }
                                viewBox="0 0 24 24"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                                  key="0"
                                />
                                <path
                                  d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                                  key="1"
                                />
                              </svg>
                            </IconBase>
                          </FiEdit>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </Button>
                </ButtonIconUpdate>
                 
                <ButtonIconDelete
                  execute={[Function]}
                >
                  <Button
                    active={false}
                    disabled={false}
                    onClick={[Function]}
                    size="sm"
                    variant="danger"
                  >
                    <button
                      className="btn btn-danger btn-sm"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <AiFillDelete>
                        <IconBase
                          attr={
                            Object {
                              "viewBox": "0 0 1024 1024",
                            }
                          }
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 1024 1024"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                              key="0"
                            />
                          </svg>
                        </IconBase>
                      </AiFillDelete>
                    </button>
                  </Button>
                </ButtonIconDelete>
              </td>
            </tr>
            <tr
              key="4"
            >
              <td>
                CNHi
              </td>
              <td>
                Não Informado
              </td>
              <td>
                80.783.926/0001-30
              </td>
              <td>
                servicos.rental@brasif.com.br
              </td>
              <td>
                <ButtonIconUpdate
                  linkTo="/urlFake/update/4"
                >
                  <Button
                    active={false}
                    as={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "Link",
                        "propTypes": Object {
                          "innerRef": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    size="sm"
                    to="/urlFake/update/4"
                    variant="primary"
                  >
                    <Link
                      className="btn btn-primary btn-sm"
                      disabled={false}
                      to="/urlFake/update/4"
                    >
                      <LinkAnchor
                        className="btn btn-primary btn-sm"
                        disabled={false}
                        href="/urlFake/update/4"
                        navigate={[Function]}
                      >
                        <a
                          className="btn btn-primary btn-sm"
                          disabled={false}
                          href="/urlFake/update/4"
                          onClick={[Function]}
                        >
                          <FiEdit>
                            <IconBase
                              attr={
                                Object {
                                  "fill": "none",
                                  "stroke": "currentColor",
                                  "strokeLinecap": "round",
                                  "strokeLinejoin": "round",
                                  "strokeWidth": "2",
                                  "viewBox": "0 0 24 24",
                                }
                              }
                            >
                              <svg
                                fill="none"
                                height="1em"
                                stroke="currentColor"
                                strokeLinecap="round"
                                strokeLinejoin="round"
                                strokeWidth="2"
                                style={
                                  Object {
                                    "color": undefined,
                                  }
                                }
                                viewBox="0 0 24 24"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                                  key="0"
                                />
                                <path
                                  d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                                  key="1"
                                />
                              </svg>
                            </IconBase>
                          </FiEdit>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </Button>
                </ButtonIconUpdate>
                 
                <ButtonIconDelete
                  execute={[Function]}
                >
                  <Button
                    active={false}
                    disabled={false}
                    onClick={[Function]}
                    size="sm"
                    variant="danger"
                  >
                    <button
                      className="btn btn-danger btn-sm"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <AiFillDelete>
                        <IconBase
                          attr={
                            Object {
                              "viewBox": "0 0 1024 1024",
                            }
                          }
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 1024 1024"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                              key="0"
                            />
                          </svg>
                        </IconBase>
                      </AiFillDelete>
                    </button>
                  </Button>
                </ButtonIconDelete>
              </td>
            </tr>
          </tbody>
        </table>
      </ForwardRef>
    </CompanyList>
  </Router>
</Provider>
`;
